test/suites/test_push.py

```diff
@@ -0,0 +1,449 @@
+)
+from .lib.tarantool_server import TarantoolServer
+from .lib.skip import skip_or_run_push_test_via_conn_pool
+
+
+def create_server():
+    srv = TarantoolServer()
+    srv.script = 'test/suites/box.lua'
+    srv.start()
+    srv.admin("box.schema.user.create('test', {password = 'test', " +
+              "if_not_exists = true})")
+    srv.admin("box.schema.user.grant('test', 'read,write,execute', 'universe')")
+
+    # Create server_function (for testing purposes).
+    srv.admin("""
+    function server_function()
+        x = {0,0}
+        while x[1] < 3 do
+            x[1] = x[1] + 1
+            box.session.push(x)
+        end
+        return x
+    end
+    """)
+
+    # Create tester space and on_replace trigger (for testing purposes).
+    srv.admin("""
+    box.schema.create_space(
+        'tester', {
+        format = {
+            {name = 'id', type = 'unsigned'},
+            {name = 'name', type = 'string'},
+        }
+    })
+    """)
+    srv.admin("""
+    box.space.tester:create_index(
+        'primary_index', {
+        parts = {
+            {field = 1, type = 'unsigned'},
+        }
+    })
+    """)
+    srv.admin("""
+    box.space.tester:create_index(
+        'primary_index', {
+        parts = {
+            {field = 1, type = 'unsigned'},
+        }
+    })
+    """)
+    srv.admin("""
+    function on_replace_callback()
+        x = {0,0}
+        while x[1] < 300 do
+            x[1] = x[1] + 100
+            box.session.push(x)
+        end
+    end
+    """)
+    srv.admin("""
+    box.space.tester:on_replace(
+        on_replace_callback
+    )
+    """)
+
+    return srv
+
+
+# Callback for on_push arg (for testing purposes).
+def push_callback(data, on_push_ctx=[]):
+                data[0][1] = data[0][1] + 1
+                on_push_ctx.append(data)
+
+
+@unittest.skipIf(sys.platform.startswith("win"),
+                 'Mesh tests on windows platform are not supported')
+class TestSuite_Push(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(self):
+        print(' PUSH '.center(70, '='), file=sys.stderr)
+        print('-' * 70, file=sys.stderr)
+
+    def setUp(self):
+        # Create server and extract helpful fields for tests.
+        self.srv = create_server()
+        self.host = self.srv.host
+        self.port = self.srv.args['primary']
+
+    def test_00_00_push_call_via_connect(self):
+        def assert_push_call(conn):
+            callback_res = []
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', category=NetworkWarning)
+                resp = conn.call(
+                    'server_function',
+                    on_push=push_callback, 
+                    on_push_ctx=callback_res
+                )
+            self.assertEqual(callback_res, [[[1, 1]], [[2, 1]], [[3, 1]]])
+            self.assertEqual(resp.data[0], [3, 0])
+
+        conn = tarantool.connect(host=self.host, port=self.port, user='test', password='test')
```

Comment:
Why not open it in `setUp`?

Reply:
I tried to do something like this:

```python
@skip_or_run_conn_pool_test
def setUp(self):
        # Open connection and connection pool to instance.
        self.conn = tarantool.connect(host=self.host, port=self.port, 
                                      user='test', password='test')
        self.conn_pool = tarantool.ConnectionPool([{'host':self.host, 'port':self.port}], 
                                                     user='test', password='test')
```
But then I didn't get how to make a skip only with `tarantool.ConnectionPool` for `3.6`

Reply:
I think you may try `if hasattr(tarantool, 'ConnectionPool')` for this one. One more advice: if ConnectionPool is not supported, better set `self.conn_pool = None` and then check `if self.conn_pool is not None` on teardown, because, for some reason, `hasattr(self, 'conn_pool')` not works well in tests.

You may also wait for https://github.com/tarantool/tarantool-python/pull/245 (I'm not sure if it's closer to merging than your PR since I don't know how's the progress on https://github.com/tarantool/tarantool-python/pull/247#discussion_r1003155010) since it also contains backport of ConnectionPool to 3.6 and you don't need any skips anymore.

Reply:
By the way, I prefer using `tarantool.Connection` instead of `tarantool.connect` and thinking of deprecating the latter one. It would be nice if you've used it for consistency with modern tests.

Reply:
Yeah, you don't need it anymore, see https://github.com/tarantool/tarantool-python/pull/247#issuecomment-1296784359